NETProvider/src/FirebirdSql.Data.FirebirdClient/FirebirdClient/FbParameter.cs

```diff
@@ -459,9 +470,12 @@ private void SetFbDbType(object value)
 
 		internal static string NormalizeParameterName(string parameterName)
 		{
-			return !string.IsNullOrEmpty(parameterName) && !parameterName.StartsWith("@")
-				? string.Format("@{0}", parameterName)
-				: parameterName;
+			if (string.IsNullOrEmpty(parameterName) || parameterName[0] == '@')
+			{
+				return parameterName;
+			}
+
+			return parameterName.PadLeft(parameterName.Length + 1, '@');
```

Comment:
Indeed, but it is fastest version of string Prepend. string.Concat works slower.
Though NormalizeParameterName now called less frequently it can be replaced with anything.